### PR TITLE
Update too DataTable Pagination

### LINKED ISSUE #8516: Data tables pagination not needed when only one page
With the DataTables plugin, when only one page of results is returned, the pagination should be hidden. It's confusing to see a big "1" button that doesn't even do anything.

Screenshot below is from http://wet-boew.github.io/v4.0-ci/demos/tables/tables-en.html with Firefox on Windows:

screenshot_2018-11-19 tables - working examples - web experience toolkit

### PR DESCRIPTION
**Pull request by ghost**
Update DataTable pagination so only appears if more than one page. Fixes #8516.

### CODE REVIEW

File: src/plugins/tables/tables.js [hunk 1]
```diff
@@ -135,31 +135,42 @@ $document.on( "init.dt draw.dt", selector, function( event, settings ) {
...
+		$( ".dataTables_paginate" ).attr( "aria-hidden", "true" );
+	} else {
+
+		// Make sure Pagination is visible
+		$( ".dataTables_paginate" ).removeClass( "hidden" );
+		$( ".dataTables_paginate" ).attr( "aria-hidden", "false" );
+
+		// Update Pagination List
+		for ( var i = 0; i < paginate_buttons.length; i++ ) {
+			var item = li.cloneNode( true );
```
Comment (duboisp) [inline src/plugins/tables/tables.js]: This are might be not supported in IE11, we will need to do some testing

File: src/plugins/tables/tables.js [hunk 2]
```diff
@@ -135,31 +135,42 @@ $document.on( "init.dt draw.dt", selector, function( event, settings ) {
...
-	// Update Pagination List
-	for ( var i = 0; i < paginate_buttons.length; i++ ) {
-		var item = li.cloneNode( true );
-		item.appendChild( paginate_buttons[ i ] );
-		ol.appendChild( item );
-	}
+	// Determine if Pagination required
+	if ( paginate_buttons.length < 4 ) {
+		$( ".dataTables_paginate" ).addClass( "hidden" );
+		$( ".dataTables_paginate" ).attr( "aria-hidden", "true" );
```
Comment (duboisp) [inline src/plugins/tables/tables.js]: Using the attribute ```aria-hidden``` is not needed because the CSS is an Accessibility-Supported Web Technology

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/tables/tables.js (hunk 1 above)
Comment (thekodester) [reply src/plugins/tables/tables.js]: This is not my code and is the existing code of WET. I just see lines 147-148.

File: src/plugins/tables/tables.js (hunk 2 above)
Comment (thekodester) [reply src/plugins/tables/tables.js]: Okay that can easily be removed.

**Comment by thekodester** [comment]
Only stuff I added was:

'''
if ( paginate_buttons.length < 4 ) {
 		$( ".dataTables_paginate" ).addClass( "hidden" );
 		$( ".dataTables_paginate" ).attr( "aria-hidden", "true" );
} else {
 		$( ".dataTables_paginate" ).addClass( "hidden" );
 		$( ".dataTables_paginate" ).attr( "aria-hidden", "true" );

 		...WET CODE...
}
'''

**Comment by thekodester** [comment]
All `aria-hidden` has been remove.

**Comment by thekodester** [comment]
Only stuff I added was:

```
if ( paginate_buttons.length < 4 ) {
 		$( ".dataTables_paginate" ).addClass( "hidden" );
} else {
 		$( ".dataTables_paginate" ).addClass( "hidden" );

 		...WET CODE...
}
```

**Comment by duboisp** [comment]
Approved at the technical review meeting. This PR would be merged after a corresponding PR with the documentation will be ready. I will send a new PR with the corresponding documentation required for this PR.

**Review by duboisp — APPROVED** [review]

**Comment by duboisp** [comment]
@GormFrank Can you write the docs and add a working example?

File: src/plugins/tables/tables.js [hunk 3]
```diff
@@ -135,31 +135,40 @@ $document.on( "init.dt draw.dt", selector, function( event, settings ) {
...
 
-	// Update Pagination List
-	for ( var i = 0; i < paginate_buttons.length; i++ ) {
-		var item = li.cloneNode( true );
-		item.appendChild( paginate_buttons[ i ] );
-		ol.appendChild( item );
-	}
+	// Determine if Pagination required
+	if ( paginate_buttons.length < 4 ) {
+		$( ".dataTables_paginate" ).addClass( "hidden" );
```
Comment (GormFrank) [inline src/plugins/tables/tables.js]: This line will remove pagination for all the datatables on the page. 

The following would fix the issue:

``` pagination.addClass( "hidden" ); ```

File: src/plugins/tables/tables.js [hunk 4]
```diff
@@ -135,31 +135,40 @@ $document.on( "init.dt draw.dt", selector, function( event, settings ) {
...
+	} else {
+
+		// Make sure Pagination is visible
+		$( ".dataTables_paginate" ).removeClass( "hidden" );
```
Comment (GormFrank) [inline src/plugins/tables/tables.js]: ``` pagination.removeClass( "hidden" ); ```

**Review by GormFrank — CHANGES_REQUESTED** [review]

**Comment by thekodester** [comment]
There also an edge case that needs to be test, if someone has custom pagination enable where next/previous is disabled. Not sure if by default DataTables is just hiding or removing them.

**Comment by thekodester** [comment]
Edge case now accounted for. And @GormFrank your changes requested where made.

**Review by GormFrank — APPROVED** [review]
